### PR TITLE
align terms and conditions checkbox correctly

### PR DESCRIPTION
**Pull request by torhoehn**
### Description (*)

This PR will correct the alignment of the terms & conditions checkbox in Checkout.

### Fixed Issues (if relevant)

1. magento/magento2#25801: Alignment of terms & conditions checkbox

### Manual testing scenarios (*)

1. Create and activate Terms & Conditions under Stores -> Settings -> Terms and Conditions with longer checkbox text
2. Enable Terms & Conditions under Stores -> Configuration -> Sales -> Checkout -> Checkout Options
3. On a mobile device go to the second step of the Checkout

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less [hunk 1]
```diff
@@ -11,6 +11,34 @@
     .checkout-agreements-block {
         .checkout-agreements {
             margin-bottom: @indent__base;
+            .checkout-agreement {
+                display: flex;
+
+                &.required > .label {
+                    &::after {
```
Comment (dmytro-ch) [inline app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less]:
```suggestion
                    &:after {
```

File: app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less [hunk 2]
```diff
@@ -11,6 +11,34 @@
...
+                        content: '';
+                    }
+                    .action.action-show {
+                        &:hover, &:active {
+                            &::after {
```
Comment (dmytro-ch) [inline app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less]:
```suggestion
                            &:after {
```

File: app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less [hunk 3]
```diff
@@ -11,6 +11,34 @@
...
+                                display: inline-block;
+                                text-decoration: none;
+                            }
+                        }
+                        &::after {
```
Comment (dmytro-ch) [inline app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less]:
```suggestion
                        &:after {
```

File: app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less [hunk 4]
```diff
@@ -11,6 +11,34 @@
...
+                            content: '*';
+                            font-size: @form-field-label-asterisk__font-size;
```
Comment (dmytro-ch) [inline app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less]: Could you please use `.lib-font-size` mixin here?

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hi @torhoehn, thank you for your contribution!
Could you please also fix the minor issues reported by Static Test?

File: app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less [hunk 5]
```diff
@@ -11,6 +11,37 @@
...
+                            &:after {
+                                display: inline-block;
+                                text-decoration: none;
+                            }
+                        }
+
+                        &:after {
+                            color: @form-field-label-asterisk__color;
+                            content: '*';
+                            font-size: .lib-font-size(@form-field-label-asterisk__font-size);
```
Comment (dmytro-ch) [inline app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less:35]:
```suggestion
                            .lib-font-size(@form-field-label-asterisk__font-size);
```

Comment (dmytro-ch) [reply app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/checkout/_checkout-agreements.less:35]: Also, the mixins calls should go first, e.g. 
```
&:after {
    .lib-font-size(@form-field-label-asterisk__font-size);
    color: @form-field-label-asterisk__color;
    content: '*';
    margin: @form-field-label-asterisk__margin;
}
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
@torhoehn, thank you for the updates.
Unfortunately, there are still failing functional tests. I'm not sure exactly, but the issue might be caused by flexbox usage.
Could you please check the test failures?

**Comment by VladimirZaets** [comment]
@torhoehn, I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Pull request closed without merging**